Bug 2060542 - Update Experimenter docs for Firefox Labs on Android (#837)
mozilla/experimenter-docs: docs/workflow/firefox-labs.mdx

## Is Firefox Labs right for what I'm building?

> Labs is integrated into Nimbus (**Desktop** and **Android** for now). In the product lifecycle, Labs fits into the Incubation/Early Stage phase: if a product team is developing a new feature, Labs can serve as the first stage of product development through Nimbus.

Reviewer: ```suggestion
Labs is integrated into Nimbus on **Firefox Desktop** and **Firefox for Android**. In the product lifecycle, Labs fits into the Incubation/Early Stage phase: if a product team is developing a new feature, Labs can serve as the first stage of product development through Nimbus.
```

> ### 6. Set things up for the Nimbus team

Reviewer: I went through here and added a bunch of formatting suggestions to do basically
```
* toplevel
  * Firefox Desktop:
  * Firefox for Anrdoid*

```

but it occurs to me that we should use the <Tabs> component provided by docusaurus

https://docusaurus.io/docs/next/markdown-features/tabs#syncing-tab-choices

Author: Thanks, made tabs!

- **Add your feature to the Nimbus feature manifest.** You can frontload this entry with placeholders if the feature work isn't complete. See [Feature Definition](/technical-reference/feature-definition) for details.

> - **Land the title and description strings** for the feature: on Desktop, add Fluent IDs to [`features.ftl`](https://searchfox.org/mozilla-central/source/toolkit/locales/en-US/toolkit/firefoxlabs/features.ftl) ([Example bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1950407)); on Android, add Android string resources to [`strings.xml`](https://searchfox.org/mozilla-central/source/mobile/android/fenix/app/src/main/res/values/strings.xml).

Reviewer: ```suggestion
- **Land the title and description strings** for the feature:
  - **Firefox Desktop**: add Fluent IDs to [`features.ftl`](https://searchfox.org/mozilla-central/source/toolkit/locales/en-US/toolkit/firefoxlabs/features.ftl) ([Example bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1950407));
  - **Firefox for Android**: add Android string resources to [`strings.xml`](https://searchfox.org/mozilla-central/source/mobile/android/fenix/app/src/main/res/values/strings.xml).
```

>   - **(Optional)** On Desktop, if you wish to add a Connect post link placeholder such as hyperlinked "Share feedback" text in the description, include an `<a>` with no `href` in the string. On Android, the link text is currently locked to "Share feedback", so no strings are required, but the Connect URL may still be added.

Author: >but the Connect URL may still be added.

ToDo: Change this, I got feedback that this portion read like the URL needs to land in tree, which is not right.

>   - **(Optional)** On Desktop, if you wish to add a Connect post link placeholder such as hyperlinked "Share feedback" text in the description, include an `<a>` with no `href` in the string. On Android, the link text is currently locked to "Share feedback", so no strings are required, but the Connect URL may still be added.

Reviewer: ```suggestion
  - **(Optional)** Adding feedback links:
    - **Firefox Desktop **: if you wish to add a Connect post link placeholder such as hyperlinked "Share feedback" text in the description, include an `<a>` with no `href` in the string. [Example](https://searchfox.org/firefox-main/rev/ef112631e252adf60d44b980d46da56718f576f8/toolkit/locales/en-US/toolkit/firefoxlabs/features.ftl#114)
    - **Firefox for Android**: This step is not required.
```

6. **Check the "Is this a Firefox Labs rollout?" checkbox** on the Branches page. This will reveal several additional fields:

   - **Title (Fluent ID)** *(required)* — The Fluent ID for your feature's title in Labs (e.g., the ID you landed in `features.ftl` on Desktop or the slug ID in `strings.xml` for Android). Firefox resolves this to localized text in `about:preferences#experimental` on Desktop or the Firefox Labs entry on Android.

>    - **Description (Fluent ID)** *(required)* — The Fluent ID for your feature's description.

Reviewer: ```suggestion

   - **The title to display in Firefox Labs** *(required)*:
     This text we be resolved to localized text (in `about:preferences#experimental` on Desktop or the *Firefox Labs* screen on Firefox for Android).
     - **Firefox Desktop**: The Fluent ID for your feature's title in Labs e.g., the ID you landed in [`features.ftl`](https://searchfox.org/firefox-main/source/toolkit/locales/en-US/toolkit/firefoxlabs/features.ftl) ([example](https://searchfox.org/firefox-main/rev/ef112631e252adf60d44b980d46da56718f576f8/toolkit/locales/en-US/toolkit/firefoxlabs/features.ftl#97). 
     - **Firefox for Android**: the slug ID in [`static_strings.xml`](https://searchfox.org/firefox-main/rev/272c6937c93bf6c3f132ce962d9ad7d798fce830/mobile/android/fenix/app/src/main/res/values/static_strings.xml) ([example](https://searchfox.org/firefox-main/rev/272c6937c93bf6c3f132ce962d9ad7d798fce830/mobile/android/fenix/app/src/main/res/values/static_strings.xml#325)).
   - **Description (Fluent ID)** *(required)*: The Fluent ID for your feature's description.
     - **Firefox Desktop**: The Fluent ID for your feature's description that the ID you landed in [`features.ftl`](https://searchfox.org/firefox-main/source/toolkit/locales/en-US/toolkit/firefoxlabs/features.ftl) ([example](https://searchfox.org/firefox-main/rev/ef112631e252adf60d44b980d46da56718f576f8/toolkit/locales/en-US/toolkit/firefoxlabs/features.ftl#99))
     - **Firefox for Android**: the slug ID in [`static_strings.xml`](https://searchfox.org/firefox-main/rev/272c6937c93bf6c3f132ce962d9ad7d798fce830/mobile/android/fenix/app/src/main/res/values/static_strings.xml) ([example](https://searchfox.org/firefox-main/rev/272c6937c93bf6c3f132ce962d9ad7d798fce830/mobile/android/fenix/app/src/main/res/values/static_strings.xml#326))
```

Author: Updated this to be in a tab! 

Heads-up for future info: `static_strings` is only for non-localized stuff, like test and strings awaiting content review. (Those strings never went through localization since they were just for QA.)

   - **Description Links (JSON)** *(optional)* — A JSON object that maps link names to URLs, enabling clickable links inside your description. For example, if your Fluent description string contains `<a data-l10n-name="connect-link">Share feedback</a>`, provide:
     ```json
     {

>        "connect-link": "https://connect.mozilla.org/t5/your-post/..."

Reviewer: mention that labs on android that "feedback" is the only supported connect link

Author: Good idea!

     | Developer Tools | 137 |
     | Productivity | 143 |

     New groups can be added, but they must ride the release trains — a new group won't be available on Release until the version it was added in reaches Release. Groups are not available on Android.

   - **Requires restart** *(optional)* — Check this if the feature requires a Firefox restart to take effect after the user opts in.

:::note

> When you add your feature to the Nimbus feature manifest, you can pull in the latest changes immediately from `FeatureManifest.yaml` or `nimbus.fml.yaml`. However, automated Experimenter syncs with the latest Nightly updates [require approval](https://github.com/mozilla/experimenter/pull/12274) before features show up in the Experimenter feature config. Once approved, the change should be available within an hour.

Reviewer: ```suggestion
When you add your feature to the Nimbus feature manifest, you can pull in the latest changes immediately from `FeatureManifest.yaml` (Firefox Desktop) or `nimbus.fml.yaml` (Firefox for Android). However, automated Experimenter syncs with the latest Nightly updates [require approval](https://github.com/mozilla/experimenter/pull/12274) before features show up in the Experimenter feature config. Once approved, the change should be available within an hour.
```

Can we also add searchfox links here?
